services/web/client/source/resource/form/service.json

```diff
@@ -0,0 +1,186 @@
+                    "type": "string",
+                    "title": "Email"
+                },
+                "affiliation": {
+                    "type": "string",
+                    "title": "Affiliation"
+                }
+            }
+        },
+        "serviceInfo": {
+            "type": "object",
+            "title": "Service information",
+            "required": [
+                "description", 
+                "commandLine"
+            ],
+            "properties": {
+                "description": {
+                    "type": "string",
+                    "title": "Description of the service",
+                    "description": "Please provide with detailed information about how to run the service and any dependencies required."
+                },
+                "commandLine": {
+                    "type": "string",
+                    "title": "Command line interface to run service",
+                    "description": "In the command line, what command is used to run the code? E.g. In Matlab Command Window: 'myproject(input1, input2, input3)'"
+                },
+                "url": {
+                    "type": "string",
+                    "title": "URL of the repository"
+                },
+                "codeVariant": {
+                    "type": "string",
+                    "title": "Language/Code type",
+                    "description": "Choose one of the accepted options.",
+                    "enum": ["matlab", "opencor", "python"]
+                },
+                "files": {
+                    "type": "string",
+                    "title": "File upload",
+                    "description": "If your code doesn't live inside a repository, you can upload a single .zip file containing it. Maximum size is 10MB",
+                    "contentMediaType": "application/zip"
+                }
+            }
+        },
+        "serviceInterface": {
+            "type": "object",
+            "title": "Service inputs and outputs",
+            "properties": {
+                "inputs": {
+                    "title": "Inputs",
+                    "type": "array",
+                    "items": {
+                        "type": "object",
+                        "title": "input",
+                        "required": [
+                            "name",
+                            "type",
+                            "label",
+                            "defaultValue",
+                            "description"
+                        ],
+                        "properties": {
+                            "name": {
+                                "type": "string",
+                                "title": "Name",
+                                "description": "Input's ID (variable name)."
+                            },
+                            "type": {
+                                "type": "string",
+                                "title": "Type"
```

Comment:
for the types we keep them open? I'm ok with this but just know that at the moment we have some restrictions on the service side (that can of course be lifted)

Reply:
that is on you. if you want to have them restricted, that is already possible using enums. send me the types or PR to this branch with the modified schema. let me know! @sanderegg

Reply:
```
"type": {
                            "type": "string",
                            "pattern": "^(number|integer|boolean|string|data:([^/\\s,]+/[^/\\s,]+|\\[[^/\\s,]+/[^/\\s,]+(,[^/\\s]+/[^/,\\s]+)*\\]))$",
                            "description": "data type expected on this input glob matching for data type is allowed",
                            "examples": [
                                "number",
                                "boolean",
                                "data:*/*",
                                "data:text/*",
                                "data:[image/jpeg,image/png]",
                                "data:application/json",
                                "data:application/json;schema=https://my-schema/not/really/schema.json",
                                "data:application/vnd.ms-excel",
                                "data:text/plain",
                                "data:application/hdf5",
                                "data:application/edu.ucdavis@ceclancy.xyz"
                            ]
                        },
```
from [https://github.com/ITISFoundation/osparc-simcore/blob/fbb5d1fad63267de49b7257c2b86dc59cfe13028/api/specs/common/schemas/node-meta-v0.0.1.json#L194-L211](https://github.com/ITISFoundation/osparc-simcore/blob/fbb5d1fad63267de49b7257c2b86dc59cfe13028/api/specs/common/schemas/node-meta-v0.0.1.json#L194-L211)

Reply:
on second thought let's keep it open. we'll have to implement missing types anyway.